snap/quota/resources.go

```diff
@@ -85,6 +95,13 @@ func (qr *Resources) validateCPUSetQuota() error {
 	if len(qr.CPUSet.CPUs) == 0 {
 		return fmt.Errorf("cpu-set quota must not be empty")
 	}
+	if cgroupVerErr != nil {
+		return cgroupVerErr
+	}
+	if cgroupVer < 2 {
```

Comment:
this is a convenient place for this but afaict this code is also called by Group.validate which is used also by ResolveCrossReferences, and that's a strange place to do this check from.  

I think what would be better is to have a separate Resources.CheckFeatureRequiremens() error method do this kind of checks, and call it from  UpdateQuotaLimits after the call to validate.

Reply:
Thanks, I added this method and call it in servicestate for CreateQuota and UpdateQuota now.